futures/src/channel.rs

```diff
@@ -481,53 +481,53 @@ impl<T: AsyncRead+AsyncWrite+Send+'static> Channel<T> {
+    pub fn close(&self, code: u16, message: &str) -> impl Future<Item = (), Error = io::Error> + Send {
         let channel_id = self.id;
         let message = message.to_string();
 
-        Box::new(self.run_on_locked_transport("close", "Could not close channel", move |transport| {
+        self.run_on_locked_transport("close", "Could not close channel", move |transport| {
             transport.conn.channel_close(channel_id, code, message, 0, 0).map(|_| None)
-        }).map(|_| ()))
+        }).map(|_| ())
     }
 
     /// ack a channel close
-    pub fn close_ok(&self) -> Box<Future<Item = (), Error = io::Error> + Send + 'static> {
+    pub fn close_ok(&self) -> impl Future<Item = (), Error = io::Error> + Send {
         let channel_id = self.id;
 
-        Box::new(self.run_on_locked_transport("close_ok", "Could not ack closed channel", move |transport| {
+        self.run_on_locked_transport("close_ok", "Could not ack closed channel", move |transport| {
             transport.conn.channel_close_ok(channel_id).map(|_| None)
-        }).map(|_| ()))
+        }).map(|_| ())
     }
 
     /// update a channel flow
-    pub fn channel_flow(&self, options: ChannelFlowOptions) -> Box<Future<Item = (), Error=io::Error> + Send + 'static> {
+    pub fn channel_flow(&self, options: ChannelFlowOptions) -> impl Future<Item = (), Error=io::Error> + Send {
         let channel_id = self.id;
 
-        Box::new(self.run_on_locked_transport("channel_flow", "Could not update channel flow", move |transport| {
+        self.run_on_locked_transport("channel_flow", "Could not update channel flow", move |transport| {
             transport.conn.channel_flow(channel_id, options.active).map(|_| None)
-        }).map(|_| ()))
+        }).map(|_| ())
     }
 
     /// ack an update to a channel flow
-    pub fn channel_flow_ok(&self, options: ChannelFlowOptions) -> Box<Future<Item = (), Error=io::Error> + Send + 'static> {
+    pub fn channel_flow_ok(&self, options: ChannelFlowOptions) -> impl Future<Item = (), Error=io::Error> + Send {
         let channel_id = self.id;
 
-        Box::new(self.run_on_locked_transport("channel_flow_ok", "Could not ack update to channel flow", move |transport| {
+        self.run_on_locked_transport("channel_flow_ok", "Could not ack update to channel flow", move |transport| {
             transport.conn.channel_flow_ok(channel_id, options.active).map(|_| None)
-        }).map(|_| ()))
+        }).map(|_| ())
     }
 
-    fn run_on_locked_transport_full<Action, Finished>(&self, method: &str, error: &str, action: Action, finished: Finished, payload: Option<(Vec<u8>, BasicProperties)>) -> Box<Future<Item = Option<RequestId>, Error = io::Error> + Send + 'static>
+    fn run_on_locked_transport_full<Action, Finished>(&self, method: &str, error: &str, action: Action, finished: Finished, payload: Option<(Vec<u8>, BasicProperties)>) -> impl Future<Item = Option<RequestId>, Error = io::Error> + Send
```

Comment:
Can you leave this one unchanged with the Box? I have a few changes locally which will allow making it impl trait without the Either stuff

Reply:
Do you want to just remove `Either` or to leave `Box` in function signature? 

If second, we have to box `wait_for_answer` result as well

Reply:
Just leave the Boxing for those two for now then please, it'll go away just after :)